### ISSUE TITLE
classgraph-4.8.130 introduces NPE when scanning with overridden classloader

### ISSUE THREAD

**Issue by limbic-derek** [classgraph/classgraph#639]
I tracked down this regression to this particular release.

Here's what I'm doing:

```java
  private void getClassInfo(Class<?> clazz) {
    try (var results =
        classGraph
            .acceptClasses(clazz.getName())
            .overrideClassLoaders(clazz.getClassLoader())
            .scan()) {
      var classInfo = results.getClassInfo(clazz.getName());
      // do stuff w/ classInfo except its null here in 4.8.130
    }
  }

  public final class SomeClass {
  }

public static test() {
  getClassInfo(SomeClass.class);
}
```

Prior to release 4.8.130, I would get what I expected (ClassInfo for SomeClass), by upgrading to this version, I get a null.
Any thoughts on this? I also verified this is still broken w/ the latest release for me.

**Comment by lukehutch** [comment]
Sorry for the delay, I have been busy traveling for the last 3 weeks. 

I figured out what is happening here. You are overriding the classloader with an `AppClassLoader` instance, the module-oriented classloader for JDK 9+. This classloader supports the traditional classpath, but it doesn't make the classpath available via any public methods or fields. If you override the classloader, the classpath is ignored (this is the change that was introduced in 4.8.130, I believe) -- because normally if you're specifying a classloader to scan, you want to ignore everything else.

I think the right fix here is to check if an override classloader is `AppClassLoader`, and if so, add the contents of the system property `java.class.path` to the list of things to be scanned.

**Comment by lukehutch** [comment]
OK, this is fixed, and will be in the next release (once I catch up with fixing a few other bugs). Thank you for your report!

**Comment by lukehutch** [comment]
Released in 4.8.139. Thank you!
